source/logic/bounties.js

```diff
@@ -21,11 +21,48 @@ function setDB(database) {
  * @param {Guild} guild
  * @param {Bounty} bounty
  * @param {Company} company
- * @param {string[]} completerIds
+ * @param {GuildMember[]} completerMembers
+ * @param {string} runMode
  */
-async function addCompleters(guild, bounty, completerIds) {
+async function addCompleters({slotNumber, posterId, bountyId}, guild, completerMembers, runMode) {
+	// Validate bounty
+	let bounty = null;
+	if (bountyId) {
+		bounty = await db.models.Bounty.findByPk(bountyId,  { include: db.models.Bounty.Company });
+	} else {
+		bounty = await db.models.Bounty.findOne({ where: { userId: posterId, companyId: guild.id, slotNumber, state: "open" }, include: db.models.Bounty.Company });
+	}
+	if (!bounty) {
+		throw `You don't have a bounty in slot ${slotNumber}.`;
+	}
+	if (bounty.userId !== posterId) {
+		throw "Only the bounty poster can add completers.";
+	}
+
+	// Validate completer IDs
+	const validatedCompleterIds = [];
+	const existingCompletions = await db.models.Completion.findAll({ where: { bountyId: bounty.id, companyId: guild.id } });
+	const existingCompleterIds = existingCompletions.map(completion => completion.userId);
+	const bannedIds = [];
+	for (const member of completerMembers.filter(member => !existingCompleterIds.includes(member.id))) {
+		if (runMode === "prod" && member.user.bot) continue;
+		const memberId = member.id;
+		await db.models.User.findOrCreate({ where: { id: memberId } });
+		const [hunter] = await db.models.Hunter.findOrCreate({ where: { userId: memberId, companyId: guild.id } });
```

Comment:
```suggestion
		const [hunter] = await findOrCreateBountyHunter(memberId, guild.id);
```
Note `findOrCreateBountyHunter()` will need to be imported.

Reply:
I'll get to this one soon-ish.